p2p/net/swarm/swarm_dial.go

```diff
@@ -63,6 +65,9 @@ var (
 	// ErrGaterDisallowedConnection is returned when the gater prevents us from
 	// forming a connection with a peer.
 	ErrGaterDisallowedConnection = errors.New("gater disallows connection to peer")
+
+	// ErrQUICDraft29 is returned instead of an ErrNoTransport when attempting to dial a QUIC draft-29 address.
+	ErrQUICDraft29 = errors.New("QUIC draft-29 has been removed, QUIC (RFC 9000) is accessible with /quic-v1")
 )
```

Comment:
```suggestion
)

// ErrQUICDraft29 wraps [ErrNoTransport] and provide a more meaningful error message while
type ErrQUICDraft29 struct{}

func (ErrQUICDraft29) Error() string {
 return "QUIC draft-29 has been removed, QUIC (RFC 9000) is accessible with /quic-v1"
}

func (ErrQUICDraft29) Unwrap() error {
 return ErrNoTransport
}
```
This is enough to allow `errors.Is(ErrQUICDraft29{}, ErrNoTransport)` to return true.